### PR TITLE
[INLONG-3719][Manager] Support data_transformation feature in Inlong 

### LINKED ISSUE #3179: [Improve][Manager] Check the source name can not be the same when saving stream source
### Description

Check the source name can not be the same when saving stream source.

### InLong Component

InLong Manager

### Are you willing to submit PR?

- [X] Yes, I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://www.apache.org/foundation/policies/conduct)

### LINKED ISSUE #3719: [Feature][Manager] Support data_transformation feature
### Description

_No response_

### Use case

_No response_

### Are you willing to submit PR?

- [X] Yes, I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://www.apache.org/foundation/policies/conduct)

### PR DESCRIPTION
**Pull request by kipshi**
### Title Name: [INLONG-3719][Manager] Support data_transformation feature in Inlong 

where *XYZ* should be replaced by the actual issue number.

Fixes #3719 

### Modifications

Add stream transform node in inlong stream.

### Verifying this change

- [X] This change added tests and can be verified as follows:

  *(example:)*
  - *Added integration tests for end-to-end deployment with large payloads (10MB)*
  - *Extended integration test for recovery after broker failure*

### Documentation

  - Does this pull request introduces a new feature? (yes)

### CODE REVIEW

File: inlong-manager/manager-common/src/main/java/org/apache/inlong/manager/common/enums/Constant.java [hunk 1]
```diff
@@ -23,8 +23,6 @@
 @Deprecated
 public class Constant {
 
-    public static final Integer UN_DELETED = 0;
```
Comment (healchow) [inline inlong-manager/manager-common/src/main/java/org/apache/inlong/manager/common/enums/Constant.java:26]: Please create a new PR to commit those unused codes, or deprecated classes.

File: inlong-manager/manager-common/src/main/java/org/apache/inlong/manager/common/enums/ErrorCodeEnum.java [hunk 2]
```diff
@@ -23,6 +23,10 @@ public enum ErrorCodeEnum {
     PERMISSION_REQUIRED(2003, "The current user does not have operation authority"),
     AUTHENTICATION_REQUIRED(2004, "Authentication failed"),
 
+    ID_IS_EMPTY(101,"Primary key is empty"),
```
Comment (healchow) [inline inlong-manager/manager-common/src/main/java/org/apache/inlong/manager/common/enums/ErrorCodeEnum.java:26]: Add a blank after `,`.

**Review by gong — APPROVED** [review]

File: inlong-manager/manager-common/src/main/java/org/apache/inlong/manager/common/enums/ErrorCodeEnum.java (hunk 2 above)
Comment (kipshi) [reply inlong-manager/manager-common/src/main/java/org/apache/inlong/manager/common/enums/ErrorCodeEnum.java]: done

File: inlong-manager/manager-common/src/test/java/org/apache/inlong/manager/common/pojo/transform/TransformDefinitionTest.java [hunk 3]
```diff
@@ -0,0 +1,100 @@
...
+
+    }
+
+    @Test
+    public void testParseDeDuplicationDefinition() {
+        List<StreamField> streamFields = createStreamFields();
+        StreamField timingField = new StreamField(2, FieldType.TIMESTAMP, "event_time", null, null);
+        DeDuplicationDefinition deDuplicationDefinition = new DeDuplicationDefinition(streamFields, timingField, 100,
+                TimeUnit.MICROSECONDS, DeDuplicationStrategy.RESERVE_FIRST);
+        Gson gson = new Gson();
```
Comment (healchow) [inline inlong-manager/manager-common/src/test/java/org/apache/inlong/manager/common/pojo/transform/TransformDefinitionTest.java:53]: Maybe we can create a global Gson instance?

**Review by healchow — APPROVED** [review]
LGTM

**Review by gosonzhang — APPROVED** [review]

**Pull request merged**